### PR TITLE
Clear prefs when stopping operational patterns

### PR DESCRIPTION
**Pull request by moritzhaller**
Resolves https://github.com/brave/brave-browser/issues/19502

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by LorenzoMinto — APPROVED** [review]

File: components/brave_federated/operational_patterns.cc [hunk 1]
```diff
@@ -188,7 +194,7 @@ void OperationalPatterns::OnDeleteUploadComplete(
   if (headers)
     response_code = headers->response_code();
   if (response_code == 200) {
```
Comment (tmancey) [inline components/brave_federated/operational_patterns.cc:196]: Please use `net::HTTP_OK` from `#include "net/http/http_status_code.h"`

Comment (tmancey) [reply components/brave_federated/operational_patterns.cc:196]: nit: can we fix other magic numbers, i.e. line 185

File: components/brave_federated/operational_patterns.cc [hunk 2]
```diff
@@ -63,12 +67,15 @@ namespace brave_federated {
 OperationalPatterns::OperationalPatterns(
     PrefService* pref_service,
     scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
-    : pref_service_(pref_service), url_loader_factory_(url_loader_factory) {}
+    : pref_service_(pref_service), url_loader_factory_(url_loader_factory) {
+  DCHECK(pref_service_);
+  DCHECK(url_loader_factory);
```
Comment (tmancey) [inline components/brave_federated/operational_patterns.cc:72]: nit: `url_loader_factory_`

**Review by tmancey — APPROVED** [review]
LGTM++ (1 nit)

File: components/brave_federated/operational_patterns.cc (hunk 2 above)
Comment (moritzhaller) [reply components/brave_federated/operational_patterns.cc]: fixed

**Pull request merged**